### PR TITLE
1724 - Clean ASCWDS Workplace - convert simple PySpark functions to inline Polars

### PR DESCRIPTION
**Pull request by garypolzin**
## Description
Trello ticket [#1724](https://trello.com/c/nfxcTbtb/1724-clean-workplace-convert-simple-functions-to-be-inline)

**Polars job**
- Converted `filter_test_accounts` and `remove_duplicate_workplaces_in_raw_workplace_data` from PySpark into one Polars expression `valid_workplace_filter`
- Added inline Polars code in place of the PySpark `remove_white_space_from_nmdsid` function
- updated `withColumnRenamed` to Polars equivalent

**Validation job**
- The filtering is quite complex so I've replaced the `row_count_match` check with `rows_distinct`

## Testing
- [X] Unit tests passing
- [X] Successful [Step Function run](https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:856699698263:execution:conv-cln-wrkplace-Transform-ASCWDS-Data:1080c66d-3956-4db9-9370-8668e75f7b65)
- [X] Outputs checked in Athena
- [X] Output Screenshots added to Trello ticket

## Migration to polars checklist
- [X] Check that similar utils functions don't already exist, or if one can be created
- [X] Check that utils are being saved in the appropriate place
- [X] Used LazyFrame option for test data
- [X] Unit tests added
- [X] Docstrings added
- [X] Added comment to pyspark functions which have been migrated with the link to the new location
      `# converted to polars -> filepath.py`
- [X] Updated CHANGELOG
- [X] Code reviewed by AI
- [X] Moved Trello ticket to PR column

## Reviewer checklist
- [ ] PR solves the Trello ticket requirements
- [ ] Outputs appear reasonable and understandable
- [ ] The overall approach is appropriate and not over-engineered
- [ ] No obvious scalability, performance or interdependency concerns
- [ ] Tests appropriately cover the main behaviour/edge cases
- [ ] Naming and structure are broadly understandable and consistent
- [ ] Docstrings are sufficient for future maintenance
- [ ] Docstrings cover non-obvious behaviour

### CODE REVIEW

File: projects/_01_ingest/ascwds/tests/fargate/test_clean_ascwds_workplace.py [hunk 1]
```diff
@@ -26,7 +26,9 @@ def test_main_runs(
             self.WORKPLACE_FOR_RECONCILIATION_DESTINATION,
         )
 
-        scan_parquet_mock.assert_called_once_with(self.WORKPLACE_SOURCE)
+        scan_parquet_mock.assert_called_once_with(
+            self.WORKPLACE_SOURCE, selected_columns=job.columns_to_import
+        )
```
Comment (RoyPriceCS50) [inline projects/_01_ingest/ascwds/tests/fargate/test_clean_ascwds_workplace.py:35]: Patch, mock, call check for valid_workplace_filter()

**Review by RoyPriceCS50 — APPROVED** [review]
Looks good. Just a comment on patch, test, call check for filtering id's
